### ISSUE TITLE
Any interest in collaborating for integration into Apache Tika?

### ISSUE THREAD

**Issue by tballison** [evernote/iwana#1]
Any chance you'd be willing to push a full build into Maven Central so that we could use this in Apache Tika?

If so, would you have interest in future support/modifications?

Thank you!

See [TIKA-1966](https://issues.apache.org/jira/browse/TIKA-1966) and [TIKA-1358](https://issues.apache.org/jira/browse/TIKA-1358).

**Comment by akhaku** [comment]
Absolutely, and we'd love to have it included in Tika! Give me a few days, will hopefully find some time soon.
I don't know if we have the resources to add modifications right now, but we're happy to accept PRs!

**Comment by tballison** [comment]
Sounds great. I'm in no rush, but it would be great to add this to Tika!

I got your code working with some very slight modifications by building protobufs from protobuf descriptors dumped by [obriensp](https://github.com/obriensp/iWorkFileFormat/blob/master/Docs/index.md).

If we could just start with packaging what you have and including the protobuf .java files, I can take it from there in Tika.

And, y, I'll be more than happy to open prs. :)

**Comment by akhaku** [comment]
Just uploaded a 1.0-SNAPSHOT version of iwana at [sonatype's ossrh](https://oss.sonatype.org/content/repositories/snapshots/com/evernote/iwana/). Have to run now so can't verify that it actually works - if you have a moment do you mind poking around to make sure I did it right? If it's good, I can push up a 1.0 release soon.

Also, feel free to yell at me if I can do things better, you definitely have a lot more experience at this than I do!

**Comment by tballison** [comment]
Fantastic!  I'll take a look.  Thank you!

**Comment by tballison** [comment]
This looks good.  A few small things...

- Let's add the unit tests in PR#2
- I think I figured out why some of the text is missing from numbers files, and I'll submit another PR for that (PR #3).  It would be great if we could fix this before pushing to maven central.

Once those are done, if you're able to push to maven central, we should be good to go...lots more to do, but this will be a great start.

Thank you, again!

**Comment by akhaku** [comment]
Sorry for the delay, work's been busy. Merged in your changes from #2 - once your fix comes in, I can bump the version to 1.0 and push to central.

**Comment by tballison** [comment]
No problem at all!  There's still quite a bit I'd like to do, but this gets us off the ground.
